AnimationCodelab/finished/src/main/java/com/example/android/codelab/animation/ui/home/Home.kt

```diff
@@ -0,0 +1,702 @@
+    onRefresh: () -> Unit
+) {
+    Row(
+        modifier = Modifier
+            .heightIn(min = 64.dp)
+            .padding(16.dp),
+        verticalAlignment = Alignment.CenterVertically
+    ) {
+        Box(
+            modifier = Modifier
+                .size(48.dp)
+                .clip(CircleShape)
+                .background(amber600)
+        )
+        Spacer(modifier = Modifier.width(16.dp))
+        Text(text = stringResource(R.string.temperature), fontSize = 24.sp)
+        Spacer(modifier = Modifier.weight(1f))
+        Icon(
+            imageVector = Icons.Default.Refresh,
+            contentDescription = stringResource(R.string.refresh),
+            modifier = Modifier
+                .clickable(onClick = onRefresh)
+                .padding(12.dp)
+                .size(24.dp)
+        )
+    }
+}
+
+/**
+ * Shows the loading state of the weather.
+ */
+@Composable
+private fun LoadingRow() {
+    val infiniteTransition = rememberInfiniteTransition()
+    val alpha by infiniteTransition.animateFloat(
+        initialValue = 0f,
+        targetValue = 1f,
+        animationSpec = infiniteRepeatable(
+            animation = keyframes {
+                durationMillis = 1000
+                0.7f at 500
+            },
+            repeatMode = RepeatMode.Reverse
+        )
+    )
+    Row(
+        modifier = Modifier
+            .heightIn(min = 64.dp)
+            .padding(16.dp),
+        verticalAlignment = Alignment.CenterVertically
+    ) {
+        Box(
+            modifier = Modifier
+                .size(48.dp)
+                .clip(CircleShape)
+                .background(Color.LightGray.copy(alpha = alpha))
+        )
+        Spacer(modifier = Modifier.width(16.dp))
+        Box(
+            modifier = Modifier
+                .fillMaxWidth()
+                .height(32.dp)
+                .background(Color.LightGray.copy(alpha = alpha))
+        )
+    }
+}
+
+/**
+ * Shows a row for one task.
+ *
+ * @param task The task description.
+ * @param onRemove Called when the task is swiped away and removed.
+ */
+@Composable
+private fun TaskRow(task: String, onRemove: () -> Unit) {
+    Surface(
+        modifier = Modifier
+            .fillMaxWidth()
+            .swipeToDismiss(onRemove),
+        elevation = 2.dp
+    ) {
+        Row(
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(16.dp)
+        ) {
+            Icon(
+                imageVector = Icons.Default.Check,
+                contentDescription = stringResource(R.string.task)
+            )
+            Spacer(modifier = Modifier.width(16.dp))
+            Text(
+                text = task,
+                style = MaterialTheme.typography.body1
+            )
+        }
+    }
+}
+
+/**
+ * The modified element can be horizontally swiped away.
+ *
+ * @param onDismissed Called when the element is swiped to the edge of the screen.
+ */
+private fun Modifier.swipeToDismiss(
```

Comment:
This whole modifier needs more comments.

Reply:
Added comments.